contracts/contracts/validator-registry/middleware/README.md

```diff
@@ -104,25 +104,40 @@ For validators who proposed incorrectly as determined by the oracle, slashing mu
+
+Read more about Symbiotic slashing guarantees [here](https://docs.symbiotic.fi/core-modules/vaults#slashing).
 
 Since a permissioned oracle account invokes slashing, the mev-commit middleware contract only requires the most basic slashing interface. Hence for Vaults that use a `VetoSlasher`, the resolver is required to be disabled via `address(0)`.
 
-Upon the oracle successfully calling `slashValidators`, the middleware contract emits one of two events for each slashed validator. `ValidatorSlashed` will be emitted for slashers with `INSTANT_SLASHER_TYPE`. `ValidatorSlashRequested` will be emitted for slashers with `VETO_SLASHER_TYPE`. If the slasher type is `VETO_SLASHER_TYPE`, the oracle is responsible for calling `IVetoSlasher.executeSlash` during the execute phase, AND this call must execute on L1 prior to the oracle calling `slashValidators` again. This ensures the oracle's subsequent calls to `slashValidators` will incorporate a properly decremented slashable stake from relevant Vaults. Read more about Symbiotic veto slashing [here](https://docs.symbiotic.fi/core-modules/vaults#veto-slashing).
+Upon the oracle successfully calling `slashValidators`, the middleware contract emits one of two events for each slashed validator. `ValidatorSlashed` will be emitted for slashers with `INSTANT_SLASHER_TYPE`. `ValidatorSlashRequested` will be emitted for slashers with `VETO_SLASHER_TYPE`. If the slasher type is `VETO_SLASHER_TYPE`, the oracle is responsible for calling `MevCommitMiddleware.executeSlashes` during the execute phase, AND this call must execute on L1 prior to the oracle calling `slashValidators` again. This ensures the oracle's subsequent calls to `slashValidators` will incorporate a properly decremented slashable stake from relevant Vaults.
+
+No action is required from the oracle during the veto phase, and following the veto phase, the oracle has a static 60 minute window, during which `executeSlashes` must be called. Read more about Symbiotic veto slashing [here](https://docs.symbiotic.fi/core-modules/vaults#veto-slashing).
 
 ### Configuration of slashPeriodSeconds
 
-`slashPeriodSeconds` must be configured such that mev-commit actors can query the proposer set for the current L1 epoch, query each proposer's opt-in status from the latest finalized L1 block (two L1 epochs ago), act upon said opt-in status during the current epoch, and allow a buffer for the oracle to possibly slash opted-in validators who act incorrectly.
+`slashPeriodSeconds` must be set such that a mev-commit bidder knows all _currently opted-in_ block proposers from the current epoch, and next epoch, must deliver commitments, or are guaranteed slashable.
 
-Concretely, for validators to be slashable by the oracle, `slashPeriodSeconds` must be greater than:
+Note _currently opted-in_ in this context, means the validator is opted-in with respect to the latest finalized L1 block state, in the worst case this is two L1 epochs ago.
```

Comment:
That's strictly speaking not correct, since finalization could take longer than 2 epochs. I guess we consider everything from more than 2 epochs ago as finalized, even if it's technically not. Maybe not the right place to discuss this, but wanted to point it out.

Reply:
Is there a probabilistic/practical block threshold you'd suggest for assuming the vast majority of txes will be finalized? 

I actually didn't know finalization could take longer than 2 epochs. A proof of stake protocol should be able to provide some sort of finalization guarantee..

Reply:
Asking cause we do need to make some sort of assumption of how long finalization can take in the worst case scenario here

Reply:
There is a block tag you can query for whether the block is already finalized or not. There is no guarantee in terms of number of block/slots etc. In theory, finalization could take arbitrarily long. But in practice, it is mostly happening after the 2 epochs. Even if not technically finalized, it already is highly to reorg much earlier due to how attestions work; I think the longest reorg ever since PoS was 7 blocks. So waiting 2 epochs is plenty in practice, but theoretically not guaranteed.

Reply:
Got it, and yea we're using the finalized block query around the repo. My question here was specific to protocol design and what we can use as a practical max reorg time for determining `slashPeriodSeconds`. Seems like 2 epochs works as that practical assumption